From 22c195edb78d0914a9b09c6cdf334591b129cb62 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Thu, 12 Sep 2024 11:19:45 +0300
Subject: [PATCH] [FSW-2704] fs1: print char to indicate ppboot load

With this patch, added support to print char
over uart to know transition from boot rom to
ppboot.

Change-Id: I8fa96a385cf1fdd332b5ce15d03529fa2a28fe00
---

diff --git a/common/boot_flow.c b/common/boot_flow.c
index 7aee0bd..e671413 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -12,6 +12,7 @@
 #include <common.h>
 #include <recovery.h>
 #include <boot_flow.h>
+#include <dev_uart.h>
 
 /* These fields will be added to proper structure later */
 static enum hl_fw_manifest_type primary_manifest_idx = HL_FW_SPI_MANIFEST_0;
@@ -114,6 +115,7 @@
 			/* TODO: Report security image fw version */
 			ppboot = (void *)((uint32_t)boot_rom_get_arc_addr(desc.start_addr));
 			boot_rom_status_update(PROGRESS_PPBOOT_START);
+			boot_rom_putc('p');
 			ppboot();
 		}
 	}
@@ -154,6 +156,7 @@
 		/* TODO: Report security image fw version */
 		ppboot = (void *)((uint32_t)boot_rom_get_arc_addr(desc.start_addr));
 		boot_rom_status_update(PROGRESS_PPBOOT_START);
+		boot_rom_putc('p');
 		ppboot();
 	}
 
